src/coreclr/jit/emitarm64.cpp

```diff
@@ -11693,15 +11693,15 @@ void emitter::emitIns_Call(EmitCallType          callType,
 
 /*****************************************************************************
  *
- *  Return an encoding for the specified 'R' register used in '18' thru '17' position.
+ *  Return an encoding for the specified 'R' register used in '17' thru '16' position.
  */
 
-/*static*/ emitter::code_t emitter::insEncodeReg_R_18_to_17(regNumber reg)
+/*static*/ emitter::code_t emitter::insEncodeReg_R_17_to_16(regNumber reg)
 {
     assert(isIntegerRegister(reg));
     emitter::code_t ureg = (emitter::code_t)reg;
     assert((ureg >= 0) && (ureg <= 32));
```

Comment:
```suggestion
    assert((ureg >= 0) && (ureg <= 3));
```
?

Reply:
`psel` uses this encoding and I am not sure what register represents by 2 bits. The [manual says](https://docsmirror.github.io/A64/2023-06/psel_p_ppi.html) "Is the 32-bit name of the vector select register W12-W15, encoded in the "Rv" field.". @a74nh do you know?

Reply:
`PSEL <Pd>, <Pn>, <Pm>.<T>[<Wv>, <imm>]`

It's the `Wv` in the above. That's quite an odd instruction encoding - it's trying to fit in 4 registers, a 5bit immediate and the element size.

It's not explicitly mentioned, but `w12` would be `00`, `w13` would be `01`, etc  

So, the assert should be `(ureg >= 12) && (ureg <= 15)`.

Reply:
hoping to get it address in https://github.com/dotnet/runtime/pull/95996 to skip a CI run.